developer-docs/latest/concepts/sso.md

```diff
@@ -0,0 +1,320 @@
+                firstname: profile.given_name,
+                lastname: profile.family_name,
+              });
+            }
+          ),
+        },
+      ],
+    },
+  },
+});
+```
+
+:::
+
+::: tab Github
+
+Using: [passport-github](https://github.com/cfsghost/passport-github)
+
+```bash
+yarn add passport-github2
+```
+
+`/config/server.js`
+
+```jsx
+'use strict';
+
+const GithubStrategy = require('passport-github2');
+
+module.exports = ({ env }) => ({
+  // ...
+  admin: {
+    // ...
+    auth: {
+      // ...
+      providers: [
+        {
+          uid: 'github',
+          displayName: 'Github',
+          icon: 'https://cdn1.iconfinder.com/data/icons/logotypes/32/github-512.png',
+          createStrategy: strapi => new GithubStrategy({
+            clientID: env('GITHUB_CLIENT_ID'),
+            clientSecret: env('GITHUB_CLIENT_SECRET'),
+            scope: ['user:email'],
+            callbackURL: strapi.admin.services.passport.getStrategyCallbackURL('github'),
+          }, (accessToken, refreshToken, profile, done) => {
+            done(null, {
+              email: profile.emails[0].value,
+              username: profile.username,
+            });
+          }),
+        },
+      ],
+    },
+  },
+});
+```
+:::
+
+::: tab Discord
+
+Using: [passport-discord](https://github.com/nicholastay/passport-discord#readme)
+
+```bash
+yarn add passport-discord
+```
+
+`/config/server.js`
+
+```jsx
+"use strict";
+
+const DiscordStrategy = require("passport-discord");
+
+module.exports = ({ env }) => ({
+  // ...
+  admin: {
+    // ...
+    auth: {
+      // ...
+      providers: [
+        {
+          uid: 'discord',
+          displayName: 'Discord',
+          icon: 'https://cdn0.iconfinder.com/data/icons/free-social-media-set/24/discord-512.png',
+          createStrategy: strapi =>
+            new DiscordStrategy(
+              {
+                clientID: env('DISCORD_CLIENT_ID'),
+                clientSecret: env('DISCORD_SECRET'),
+                callbackURL: strapi.admin.services.passport.getStrategyCallbackURL(
+                  'discord'
+                ),
+                scope: ['identify', 'email'],
+              },
+              (accessToken, refreshToken, profile, done) => {
+                done(null, {
+                  email: profile.email,
+                  username: `${profile.username}#${profile.discriminator}`,
+                });
+              }
+            ),
+        },
+      ],
+    },
+  },
+});
+```
+:::
+::::
+
+## Advanced Customization
+
+### Admin Panel URL
+
+If your administration panel lives on a different host/port than your Strapi server, you will need to modify the admin URL.
+To do so, head to your `/config/server.js` configuration file and tweak the `admin.url` field.
+
+For example, if your admin application has been started on `localhost:4000`, your configuration will look like the following:
+
+```javascript
+{
+  ...
+  admin: {
+    url: 'http://localhost:4000/admin',
```

Comment:
Technically the user sets the running port to 4000, this isn't needed as it will automatically default to this. We should avoid localhost examples and instead provide simulation real world examples. IE `https://api.example.com` or something like that.

I'll try my hand with a commit to modify some of the wording.